pandas/core/missing.py

```diff
@@ -520,6 +520,26 @@ def fill_zeros(result, x, y, name, fill):
 
     Mask the nan's from x.
     """
+    if name == '__divmod__' and isinstance(result, tuple):
+        # GH#26987; sometimes we don't have a tuple when called
+        #  from dispatch_missing
+        res1 = fill_zeros(result[0], x, y, '__floordiv__', fill)
+        res2 = fill_zeros(result[1], x, y, '__mod__', fill)
+        mask = np.isinf(res1) & np.isinf(res2)
+        if mask.any():
+            res2[mask] = np.nan
+        return res1, res2
+
```

Comment:
i think it ends up being a wash because we have to do these checks in multiple places

Reply:
the style of this change is not consistent with the rest of the function, it could be better integrated, for example a helper function would go a long way here.